### ISSUE TITLE
#34195 ⁃ [a11y] Game links the World Cup widget on HNT are not exposed to assistive technology as controls

### ISSUE THREAD

**Issue by anna-yeddi** [mozilla-mobile/firefox-ios#34195]
## Steps to reproduce

1. [while #34193 exists, this step needs to come after the widget's page is changed to the "Group Stage"] Have VoiceOver running
2. Open a new tab and confirm the World Cup widget is present between the Shortcuts and News sections
3. Navigate to the widget pagination and change the page to one of the game pages, i.e. (currently 2 of 18) "Group Stage"
4. Review the rows for each game with VO (by swiping left-to-right with one finger) and observe the VO announcement
5. Use VO Rotor and attempt to get to the game links via any Rotor actions (i.e. "Unvisited links", or "Form controls", or any other option available)
6. Attempt to activate a game link and observe the VO behavior

Repeat steps with Switch Control or Voice Control.

### Expected behavior

1. The game controls are focused with VO as one row item and the announcement includes both teams playing and the time **and** the role of the control is announced, i.e. `MEX, 3:00 PM, RSA, link` or `Mexico - South Africa, 3:00 PM, button`
2. Switch Control user can focus and activate each game control
3. Voice Control users can activate the control by commanding: "Tap {on-screen text}" or, in "show numbers" mode, by commanding: "Tap {3}"

- when VC is running and the widget is in the view, the VC does show a control's name or a number by default

### Actual behavior

1. The VO focuses and announces all items within the game row as individual static text, i.e.: `MEX` > swipe > `3:00 PM` > swipe > `RSA` > swipe to the next row, no role indication is provided

- VO Rotor even when these items are focused does not provide any indication of these elements being any types of controls, when trying to navigate the page by form controls or links the game rows are skipped by VO Rotor
- Note: VO users technically *can* activate game row items, but since there is no indication it is possible (no role), *it is not expected in any way*

1. Switch Control user does not get any of the game row items highlighted in items mode

- Note: SC users technically *could* use scan mode to send a tap on a specific point on the screen, but there is no visual indication these game row items are *not* static text, thus *this is not expected*

1. Voice Control users are not getting any label/number shown for any of the games (there is no indication for them that these rows are, in fact, functional 
2. **Outcome: User cannot get to any of the game control within the widget with any of assistive technology**. Access to game information from the widget is blocked

### Recommendation

- As recommended during the WC design review, making the whole row as one focusable element and assigning it a trait of `link` or, at least, `button` is needed (while ensuring the accessibility action still activates the link - but this seems to be somewhat working now).

### Device & build information

- Device: iPhone 15 Pro running iOS 26.5 
- Build version: Firefox 151.2 (72455) 
- First seen version: ? 

### Notes

Attachments: 

[[On the left half, the time of the game is focused with VO and the output captions panel showing only on-screen text without a role; on the right half the Voice Control is not showing names for the game rows thus providing no indication these are, in fact, controls]](https://github.com/user-attachments/assets/ae6c00bf-a148-4381-a2e6-822d65313a06)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-16031)

**Comment by data-sync-user** [comment]
> **Andrei Bodea** commented from Jira:

Re-opening in v9000 (72996) with iPhone 16 (iOS 26.5), iPhone 17 Pro (26.0).
Switch control is working without issues (see the video below)

[ScreenRecording_06-23-2026 14-32-29_1.MP4](https://mozilla-hub.atlassian.net/rest/api/3/attachment/content/132556)

[ScreenRecording_06-23-2026 14-42-23_1.MP4](https://mozilla-hub.atlassian.net/rest/api/3/attachment/content/132557)



We are opening this because Voice Control is not working on iOS 26 (see the video below).
Swiping to the left and right is not working.

[ScreenRecording_06-30-2026 14-58-40_1.MP4](https://mozilla-hub.atlassian.net/rest/api/3/attachment/content/132558)
